pkg/provisioner/templates/kernel.go

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package templates
+
+import (
+	"bytes"
+	"fmt"
+	"text/template"
+
+	"github.com/NVIDIA/holodeck/api/holodeck/v1alpha1"
+)
+
+const kernelTemplate = `
+{{- if .Spec.Kernel.Modify }}
+# Install required packages for kernel compilation
+sudo apt-get update
+install_packages_with_retry build-essential libncurses-dev bison flex libssl-dev libelf-dev
+
+# Download and extract kernel source
+cd /usr/src
+{{- if .Spec.Kernel.Version }}
+KERNEL_VERSION="{{ .Spec.Kernel.Version }}"
+{{- else }}
+# Get latest stable kernel version
+KERNEL_VERSION=$(curl -s https://www.kernel.org/releases.json | grep -o '"version":"[^"]*"' | head -1 | cut -d'"' -f4)
+{{- end }}
+
+# Download and extract the kernel source
+with_retry 3 5 wget https://cdn.kernel.org/pub/linux/kernel/v${KERNEL_VERSION%%.*}.x/linux-${KERNEL_VERSION}.tar.xz
```

Comment:
Why not install using the `apt install linux-image-${KERNEL_VERSION}` command instead? That should shorten this script

Reply:
Done